### ISSUE TITLE
Warn about subshells not respecting `set -e` ins some situations

### ISSUE THREAD

**Issue by cdman** [koalaman/shellcheck#1648]
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
set -e
echo "Foo: $(false)"
echo "After"
```

#### Here's what shellcheck currently says:
No issues detected! :)

#### Here's what I wanted or expected to see:
Warn about the fact that exit values inside of echoed strings are not kept and recommend something like:

```sh
# ...
ouput=$(false)
echo "Foo: $output"
# ...
```

Which would die (as expected) if `set -e` is in effect.

**Comment by rlpowell** [comment]
I just came here to create this issue but decided to check first.  :D  So, +1 from me.  FWIW I found a good explanation/discussion at https://unix.stackexchange.com/a/23099

**Comment by koalaman** [comment]
This is now available in 0.8.0 as an optional check, see https://github.com/koalaman/shellcheck/issues/2478

**Comment by rlpowell** [comment]
Oh, cool, thank you!
